### PR TITLE
add module odgi/build

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

This PR progresses https://github.com/nf-core/pangenome/issues/70.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by subwaystation** [comment]
@mirpedrol Do you have any idea, why no test is run?

```
PROFILE=docker pytest --tag odgi/build --symlink --keep-workflow-wd
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.8, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/heumos/git/modules, configfile: pytest.ini
plugins: workflow-1.6.0, anyio-3.6.1
collecting ... 
collected 1147 items                                                                                                                                                                                                                         

tests/test_versions_yml.py sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 17%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 37%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 57%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 77%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 97%]
ssssssssssssssssssssssss                                                                                                                                                                                                               [100%]

============================================================================================================== warnings summary ==============================================================================================================
../../.local/lib/python3.10/site-packages/pytest_workflow/plugin.py:122: 673 warnings
  /home/heumos/.local/lib/python3.10/site-packages/pytest_workflow/plugin.py:122: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return YamlFile.from_parent(parent, fspath=path)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================== 1147 skipped, 673 warnings in 7.82s =====================================================================================================
```

**Comment by subwaystation** [comment]
@nvnieuwk Maybe you know why no test is actually executed?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: tests/modules/nf-core/odgi/build/main.nf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { ODGI_BUILD } from '../../../../../modules/nf-core/odgi/build/main.nf'
+
+workflow test_odgi_build {
+
+    input = [ [ id:'test' ],
+              [ file(params.test_data['homo_sapiens']['pangenome']['pangenome_smoothxg_gfa'], checkIfExists: true) ],
```
[the remark this answers is not in the document]
Comment (subwaystation) [reply tests/modules/nf-core/odgi/build/main.nf]: Unfortunately, that didn't fix the issue.

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply tests/modules/nf-core/odgi/build/main.nf]: I'll take a better look tomorrow :p

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply tests/modules/nf-core/odgi/build/main.nf]: I have no clue, I posted on Slack about it

[the remark this answers is not in the document]
Comment (subwaystation) [reply tests/modules/nf-core/odgi/build/main.nf]: Ah, you were 9 minutes faster. I always get a delay here xD

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply tests/modules/nf-core/odgi/build/main.nf]: No problem posting it twice can't hurt, right ;)

[the remark this answers is not in the document]
Comment (subwaystation) [reply tests/modules/nf-core/odgi/build/main.nf]: `build` is a keyword for `pytest` at @nvnieuwk ...... See https://nfcore.slack.com/archives/CJRH30T6V/p1666255001028999.

Could you please take a quick look again? All should be fine now.

**Comment by subwaystation** [comment]
@nvnieuwk Somehow I can't merge, although I resolved your issue. Hmm.

**Review by nvnieuwk — APPROVED** [review]

**Comment by nvnieuwk** [comment]
Hit it ;)

**Pull request merged**